libraries/rush-lib/src/api/RushConfiguration.ts

```diff
@@ -1170,21 +1157,70 @@ export class RushConfiguration {
   }
 
   /**
-   * The full path of the shrinkwrap file that is tracked by Git.  (The "rush install"
-   * command uses a temporary copy, whose path is tempShrinkwrapFilename.)
-   * @remarks
-   * This property merely reports the filename; the file itself may not actually exist.
-   * Example: `C:\MyRepo\common\npm-shrinkwrap.json` or `C:\MyRepo\common\pnpm-lock.yaml`
+   * The folder where Rush's additional config files are stored.  This folder is always a
+   * subfolder called `config\rush` inside the common folder.  (The `common\config` folder
+   * is reserved for configuration files used by other tools.)  To avoid confusion or mistakes,
+   * Rush will report an error if this this folder contains any unrecognized files.
    *
-   * @deprecated Use `getCommittedShrinkwrapFilename` instead, which gets the correct common
-   * shrinkwrap file name for a given active variant.
+   * Example: `C:\MyRepo\common\config\rush`
    */
-  public get committedShrinkwrapFilename(): string {
-    return this.getCommittedShrinkwrapFilename();
+  public getCommonRushConfigFolder(subspaceName?: string | undefined): string {
+    if (subspaceName) {
+      if (this._subspaceConfigFolderBySubspace.has(subspaceName)) {
+        return this._subspaceConfigFolderBySubspace.get(subspaceName) as string;
+      }
+      this.validateSubspaceName(subspaceName);
+
+      // If this subspace doesn't have a configuration folder, check if it is in the project folder itself
+      // if the splitWorkspaceCompatibility option is enabled in the subspace configuration
+      let subspaceConfigFolder: string = `${this.commonFolder}/config/subspaces/${subspaceName}`;
+      if (
+        !FileSystem.exists(subspaceConfigFolder) &&
```

Comment:
@william2958 testing for existence of a directory is not very reliable. For example, the folder may exist after building some previous branch but missing after a clean `git clone`.  

It would be more reliable to test for `pnpm-lock.yaml`, although we have to be careful because this file can get deleted. Maybe the logic would be something like

- If `pnpm-lock.yaml` exists in the project folder and NOT under `config/subspaces` then use splitWorkspaces
- If it exists in both places, that is an error
- If it exists in neither place, then use `config/subspaces`

A more rigorous approach might be to say that a splitWorkspace subspace name always has a `split-` prefix (e.g. `split-my-project-name`).

What do you think?

Reply:
yes I think having the split prefix would be useful, especially to easily determine which projects are still left in a split workspace mode, which would help teams migrate projects.

Changed to "split_" prefix